parser/src/lexer/compat.rs

```diff
@@ -0,0 +1,910 @@
+        ]);
+    }
+
+    #[test]
+    fn test_lex_keywords() {
+        python_tokenize_test_lexer(&[
+            "False None True and as assert async await",
+            "break class continue def del elif else except",
+            "finally for from global if import in is lambda",
+            "nonlocal not or pass raise return try while with yield",
+        ]);
+    }
+
+    #[test]
+    fn test_lex_identifiers() {
+        python_tokenize_test_lexer(&["a", "a_a", "_a", "a_", "a_a_a", "a_a_"]);
+    }
+
+    #[test]
+    fn test_lex_literals() {
+        // Binary
+        python_tokenize_test_lexer(&[
+            "0b0", "0b1", "0b10", "0b11", "0b100", "0b101", "0b110", "0b111",
+        ]);
+
+        // Octal
+        python_tokenize_test_lexer(&["0o0", "0o1", "0o2", "0o3", "0o4", "0o5", "0o6", "0o7"]);
+
+        // Hexadecimal
+        python_tokenize_test_lexer(&[
+            "0x0", "0x1", "0x2", "0x3", "0x4", "0x5", "0x6", "0x7", "0x8", "0x9", "0xa", "0xb",
+            "0xc", "0xd", "0xe", "0xf", "0xA", "0xB", "0xC", "0xD", "0xE", "0xF",
+        ]);
+
+        // Point float
+        python_tokenize_test_lexer(&["0.0 0.1 00.0 00.1 0.1j 0.01J"]);
+
+        // Exponent float
+        python_tokenize_test_lexer(&["0e0 0e-1 0e+2 0e+3j 0e+3J"]);
+
+        // Integer
+        python_tokenize_test_lexer(&["11 33 1j 1_000_000j"]);
+
+        // Strings
+        python_tokenize_test_lexer(&[
+            "\"hello\"  ",
+            "\"world\"",
+            "\"\"",
+            "a = \"hello\"",
+            "'hello'",
+            "\"\"\"hello\"\"\"",
+            "'''hello'''",
+        ]);
+
+        // Bytes
+        python_tokenize_test_lexer(&[
+            "b\"hello\"",
+            "b\"world\"",
+            "b\"\"",
+            "a = b\"hello\"",
+            "b'hello'",
+            "b\"\"\"hello\"\"\"",
+            "b'''hello'''",
+        ]);
+
+        // Raw strings
+        python_tokenize_test_lexer(&[
+            "r\"hello\"",
+            "r\"world\"",
+            "r\"\"",
+            "a = r\"hello\"",
+            "r'hello'",
+            "r\"\"\"hello\"\"\"",
+            "r'''hello'''",
+        ]);
+
+        // Raw bytes
+        python_tokenize_test_lexer(&[
+            "rb\"hello\"",
+            "rb\"world\"",
+            "rb\"\"",
+            "a = rb\"hello\"",
+            "rb'hello'",
+            "rb\"\"\"hello\"\"\"",
+            "rb'''hello'''",
+        ]);
+
+        // Unicode strings
+        python_tokenize_test_lexer(&[
+            "u\"hello\"",
+            "u\"world\"",
+            "u\"\"",
+            "a = u\"hello\"",
+            "u'hello'",
+            "u\"\"\"hello\"\"\"",
+            "u'''hello'''",
+        ]);
+    }
+
+    #[test]
+    fn test_lex_imports() {
+        python_tokenize_test_lexer(&["import a", "import a.b", "import a.b.c", "import a from b"]);
+    }
+
+    // TODO lex_python: Decide whether to keep this test or not. The Python lexer + enderpy lexer
```

Comment:
Thanks for leaving this here. I think we should do what python does because that's also what editors do so we can identify the line number correctly.